### PR TITLE
Add minimal cmake support

### PR DESCRIPTION
**Pull request by Mike-Devel**
Allows the integration of a local Boost.Integer copy into a cmake project via `add_subdirectory( libs/integer )`

### CODE REVIEW

**Comment by pdimov** [comment]
```
git clone --depth 1 https://github.com/boostorg/assert.git ../static_assert
```

**Comment by Mike-Devel** [comment]
Sorry. Will fix once I'm back on a dev machine.

**Comment by Mike-Devel** [comment]
Let's see if this passes CI. 

Btw.: Is there a special reason, why most travis scripts on boost are only run for master and develop? It is a bit annoying if you first want to test your changes on your own branch on your own travis account (which usually has a much lower delay than the boost one).

**Comment by pdimov** [comment]
I usually enable feature branches as well in my .yml files. But in case they aren't, you can modify the .yml files on your branch while testing it.

**Comment by Lastique** [comment]
A limited CMakeLists.txt is added in 1311077. Thanks.

**Comment by Lastique** [comment]
> is there a reason, you don't want the CMakeLists.txt file to be tested at all?

The test doesn't really test anything. It may still pass if there are missing dependencies from one of the headers not included by `boost/integer.hpp`.

Generally, I don't see much point in testing the build system or metadata. The dependency list may go out of sync, but I'm not sure how to test it better and not cause much maintenance annoyance. Maybe it should be generated by boostdep. @pdimov, could it?

**Comment by pdimov** [comment]
It can in principle, I had it done (`boostdep --cmake <library>`), but at the end decided against using it.

For testing I just run the tests using CMake.

**Comment by Lastique** [comment]
> I'd really prefer if boost libraries weren't so careless about adding boost-internal dependencies.

Being careful about dependencies does not imply introducing the potential for errors due to a missing dependency. Or having to manually spell and maintain the list. Even without me modifying Boost.Integer, stuff gets moved around Boost from time to time, which can make the list inactual.

> Effectively you are saying tests are worthless, because they are not complete?

I'm saying that (a) the original test was not testing what it intended to (i.e. it would not register a missing dependency) and (b) that I don't see much point in testing the build system, in general. We don't test Boost.Build in libraries, the build system has to be tested on its own.

What is worth testing and what I forsee to go out of sync easily is the dependency list. But as I said, I would prefer it to be auto-generated instead of tested. (It's not a prerequisite for PR, just a thought.)

File: test/test_cmake/main.cpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#include <boost/integer.hpp>
+
+int main() {
+
```
Comment (jeking3) [inline test/test_cmake/main.cpp]: Without adding code that the Boost.Integer link library depends on, this may not be a complete test.

Comment (jeking3) [reply test/test_cmake/main.cpp]: Nevermind, I see there's no link library. :)

**Pull request closed without merging**